src/main/java/dk/cngroup/trainings/spring/springassignment/service/AnimalService.java

```diff
@@ -2,26 +2,30 @@
 
 import dk.cngroup.trainings.spring.springassignment.model.Animal;
 import dk.cngroup.trainings.spring.springassignment.model.CareTaker;
-import dk.cngroup.trainings.spring.springassignment.service.exception.InvalidAnimalException;
-import dk.cngroup.trainings.spring.springassignment.service.exception.InvalidCareTakerException;
+import dk.cngroup.trainings.spring.springassignment.service.exception.*;
 
 import java.util.List;
 import java.util.Optional;
 
 public interface AnimalService {
 	List<Animal> getAnimals();
 
-	Optional<Animal> getAnimalById(long id);
+	Optional<Animal> getAnimalById(long id) throws AnimalNotFoundException;
 
 	List<Animal> getAnimalsByName(String name);
 
 	Optional<Animal> addAnimal(Animal animal) throws InvalidAnimalException;
 
-	boolean deleteAnimalById(long id);
+	void deleteAnimalById(long id) throws AnimalNotFoundException;
```

Comment:
because of this, tests now do not compile... always run tests before you commit anything. Actually you should run them always when you change something :)

Reply:
Yeah, tests are a pain. It's because I still don't know the final version of all my methods that I avoid refactoring tests every time.